src/playground/oauth2-password/oauth2-password-login.component.ts

```diff
@@ -0,0 +1,146 @@
+                  [required]="getConfigValue('forms.validation.password.required')"
+                  [minlength]="getConfigValue('forms.validation.password.minLength')"
+                  [maxlength]="getConfigValue('forms.validation.password.maxLength')">
+                  <small class="form-text error" *ngIf="password.invalid && password.touched"> Password is required!
+                  </small>
+                  <small class="form-text error" *ngIf="password.invalid && password.touched"> Password should contains
+                    from {{ getConfigValue('forms.validation.password.minLength') }} to
+                    {{ getConfigValue('forms.validation.password.maxLength') }} characters
+                  </small>
+                </div>
+                <button [disabled]="submitted || !form.valid" class="btn btn-block btn-hero-success"
+                        [class.btn-pulse]="submitted"> Sign In
+                </button>
+              </form>
+            </nb-auth-block>
+          </nb-card-body>
+        </nb-card>
+      </nb-layout-column>
+    </nb-layout>    `,
+})
+export class NbOAuth2PasswordLoginComponent implements OnDestroy {
+
+  token: NbAuthOAuth2Token;
+  jwtHelper = new JwtHelperService();
+
+  redirectDelay: number = 0;
+  showMessages: any = {};
+  strategy: string = '';
+
+  errors: string[] = [];
+  messages: string[] = [];
+  user: any = {};
+  submitted: boolean = false;
+
+  constructor(private authService: NbAuthService, @Inject(NB_AUTH_OPTIONS) protected options = {},
+              protected router: Router) {
+    this.redirectDelay = this.getConfigValue('forms.login.redirectDelay');
+    this.showMessages = this.getConfigValue('forms.login.showMessages');
+    this.strategy = this.getConfigValue('forms.login.strategy');
+    this.authService.onTokenChange()
+      .subscribe((token: NbAuthOAuth2Token) => {
+        this.token = null;
+        if (token && token.isValid()) {
+          this.token = token;
+        }
+      });
+  }
+
+  login(): void  {
+
+    this.errors = this.messages = [];
+    this.submitted = true;
+
+    this.authService.authenticate(this.strategy, this.user).subscribe((result: NbAuthResult) => {
+      this.submitted = false;
+      if (result.isSuccess()) {
+        this.messages = result.getMessages();
+      } else {
+        this.errors = result.getErrors();
+      }
+
+      const redirect = result.getRedirect();
+      if (redirect) {
+        setTimeout(() => {
+          return this.router.navigateByUrl(redirect);
+        }, this.redirectDelay);
+      }
+    });
+  }
+
+
+  logout() {
+    this.authService.logout('password')
+      .subscribe((authResult: NbAuthResult) => {
+      });
+  }
+
+  ngOnDestroy(): void {
+  }
+
+  getConfigValue(key: string): any {
+    return getDeepFromObject(this.options, key, null);
+  }
+
+  getClaims(rawToken: string): string {
+    return this.jwtHelper.decodeToken(rawToken);
```

Comment:
I don't think we need a new library to handle token decoding, even though this is a demo part, not framework.
Stil,l we can use our `NbAuthJWTToken` to decode token value of `NbAuthOAuth2Token`:

```
nbAuthCreateToken(NbAuthJWTToken, rawToken).getPayload();
```